### PR TITLE
Decisions details modal

### LINKED ISSUE #3708: Decisions: Create Decisions details modal (UI Only)
### Story

As a user when creating a Simple Decision, I want to be presented with a dialog that allows me to enter the decision details.

A similar component has already been implement in CoinMachine (using https://tiptap.dev/):
https://github.com/JoinColony/coinMachine/pull/125

### Design
Figma: https://www.figma.com/file/oRaTwsD5BUH3FJNzoZdSCv/Discussions?node-id=1018%3A54646

Screenshot 2022-08-02 at 12 40 42

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR tracks the work involved in creating the Decisions details modal, accessed via the New Decisions button. 

I went off the CoinMachine implementation https://github.com/JoinColony/coinMachine/pull/125, hence any discrepancies with the design. I also rendered the tooltip popover inside the form, which appears to be standard practice (at least, having it display outside like in the design would require setting the overflow to visible on the Dialog component, and since this hasn't been done, I didn't want to change it). 

I also hooked the Rich Text component into the Formik state, so it validates like a normal input. 

**New stuff** ✨

New components: 

* `NewDecisionDialogForm`
* `RichText`
* `Toolbar`

**Changes** 🏗

* `NewDecisionDialog`: added validation schema

## Screenshots 

decision-details

Resolves #3708

### CODE REVIEW

**Comment by danbr** [comment]
@willm30 Hey - unfortunately for me the dapp doesnt compile & npm is complaining.

Screenshot 2022-08-08 at 10 17 05

**Comment by willm30** [comment]
> @willm30 Hey - unfortunately for me the dapp doesnt compile & npm is complaining.
> 

@danbr 
I'm using the tiptap library like in the coinmachine implementation.

You should be able to resolve with: npm install --legacy-peer-deps

**Review by Julianahlk — APPROVED** [review]
Very nice! It works well for me. 

Screen Shot 2022-08-08 at 11 56 55 AM

As a side note, images are not fully loading and the text color cannot be changed. Might be a good things to address in another PR?

**Comment by danbr** [comment]
@willm30 package-lock.json needs to be committed.

**Comment by willm30** [comment]
> @willm30 package-lock.json needs to be committed.

@danbr Ah ok, apologies. Now pushed.

**Review by danbr — CHANGES_REQUESTED** [review]
Looking really good. Well done. 
The core functionality is there, but i've found a few things to address:

- Icon names, i made a comment in the code.

- Create Motions in..... should be 'Proposal will be created in'
Screenshot 2022-08-08 at 13 35 14

- RickText - I'm a bit confused by the name of the component. For me, I think of the file format, so i didnt find it intuitive to know its purpose. 

- Preview button is 160px in figma

- Its hard to know which icons in the toolbar have been selected. Would expect icon to be colorised/inverted.

- Toolbar icon tooltips - some show title, some show shortcuts (Github show both as a tooltip)
Screenshot 2022-08-08 at 13 38 37
Screenshot 2022-08-08 at 13 38 43

- Toolbar has several extra buttons, URL and File. Not sure they are necessary for this feature (could be useful to have in other dialogs in the future)???
Especially supporting 'files' increases complexity as we would then need to handle the provided files. 
I would suggest to keep it simple, and progress from there and add files in V2 if required.

- I think there is already a comment for this issue (i'll add just in case not). Selecting a color does not change the color of the text.

- Should we have a placeholder for Title? What think?

- Spacing between seems too large:
Screenshot 2022-08-08 at 15 18 04

- I was playing a little with the editor, and the backtick char (`) gave unexpected behaviour. This is within TipTap?

This issue has become quite large, so we can break out into smaller if you wish? 
Well done, once again.......

**Comment by willm30** [comment]
> Looking really good. Well done. The core functionality is there, but i've found a few things to address:
> 

> This issue has become quite large, so we can break out into smaller if you wish? Well done, once again.......

@danbr Thanks for the detailed feedback!

I have fixed the following:

> * Icon names, i made a comment in the code.
> * Create Motions in..... should be 'Proposal will be created in'
> * RickText - I'm a bit confused by the name of the component. For me, I think of the file format, so i didnt find it intuitive to know its purpose.

(changed to `RichTextEditor`)

> * Preview button is 160px in figma
> * Its hard to know which icons in the toolbar have been selected. Would expect icon to be colorised/inverted.
> * Toolbar icon tooltips - some show title, some show shortcuts (Github show both as a tooltip)
> * Toolbar has several extra buttons, URL and File. Not sure they are necessary for this feature (could be useful to have in other dialogs in the future)???
> * Should we have a placeholder for Title? What think?
> * Spacing between seems too large

This works for me. Can you help me reproduce?
> * I think there is already a comment for this issue (i'll add just in case not). Selecting a color does not change the color of the text.

change-color

> * I was playing a little with the editor, and the backtick char (`) gave unexpected behaviour. This is within TipTap?

This is caused by the `MuliWeb` font. It also happens in the Title input. I think we should change the font inside the RichTextEditor -- any suggestions?

File: src/modules/core/components/Button/Button.tsx [hunk 1]
```diff
@@ -44,7 +44,9 @@ export interface Props
   disabled?: boolean;
 
   /** Pass a ref to the `<button>` element */
-  innerRef?: (ref: HTMLElement | null) => void;
+  innerRef?:
+    | ((ref: HTMLElement | null) => void)
+    | React.MutableRefObject<HTMLButtonElement | null>;
```
Comment (willm30) [inline src/modules/core/components/Button/Button.tsx:49]: Updating the component to allow the passing of an actual ref object (used by the color picker), not just the function you receive when using the popover component

File: src/modules/dashboard/components/MotionDomainSelect/MotionDomainSelect.tsx [hunk 2]
```diff
@@ -46,8 +48,10 @@ const MotionDomainSelect = ({
        */
       let displayLabel =
         parseInt(option?.value || `${ROOT_DOMAIN_ID}`, 10) === ROOT_DOMAIN_ID
-          ? `${formatMessage(MSG.createDomain)} ${label}`
-          : `${formatMessage(MSG.createDomain)} ${formatMessage({
+          ? `${formatMessage(dropdownLabel || MSG.createDomain)} ${label}`
+          : `${formatMessage(
+              dropdownLabel || MSG.createDomain,
+            )} ${formatMessage({
```
Comment (willm30) [inline src/modules/dashboard/components/MotionDomainSelect/MotionDomainSelect.tsx:54]: Modifying component to accept a custom label. If I didn't do this, I feel like I would basically be copying this component into a new component with a different name for use in this form.

**Comment by danbr** [comment]
> The color picker works for me. Can you help me reproduce?
I re-checked...... and its working perfectly.



> > * I was playing a little with the editor, and the backtick char (`) gave unexpected behaviour. This is within TipTap?
> 
> This is caused by the `MuliWeb` font. It also happens in the Title input. I think we should change the font inside the RichTextEditor -- any suggestions?
> 
> I also updated the styling for code blocks (white text on black background).

Not sure how coinMachine implemented it, but the code block doesnt look pretty i.e. the black block.
I like how GitHub implements this action. But this would mean that we need Markdown in TipTap, but they dont support it...... <BIG BUT> I found MD for TipTap - https://github.com/aguingand/tiptap-markdown 
Its a new project..... Perhaps you can check if it's usable. Internally when we discussed the decisions editor, MD was spoken about. I think MD is ubiquitous these days, and a reasonable expectation from the users for it to be supported. 
Especially when things like code blocks are included.

If the MD npm isnt suitable, then i suggest to leave 'as is' and we can review this in the future.

**Review by danbr — APPROVED** [review]
Good work @willm30 - looking great. Thanks for the updates.

**Pull request merged**